core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/cumSum.kt

```diff
@@ -18,6 +21,36 @@ import kotlin.reflect.typeOf
 
 // region DataColumn
 
+/**
+ * ## The CumSum Operation
+ *
+ * Computes the cumulative sum of the values in the {@get DATA_TYPE}.
+ *
+ * __NOTE:__ If the column contains nullable values and `skipNA` is set to `true`,
+ * skips null values when computing the cumulative sum.
+ * Otherwise, any null value encountered will propagate null values in the output from that point onward.
+ *
+ * {@get [CUMSUM_PARAM] @param [columns]
+ * The names of the columns to apply cumSum operation.}
+ *
+ * @param [skipNA] Whether to skip null values (default: `true`).
+ *
+ * @return A new {@get DATA_TYPE} of the same type with the cumulative sum of the values.
+ *
+ * {@get [CUMSUM_PARAM] @see [Selecting Columns][SelectSelectingOptions].}
```

Comment:
I see you used `CUMSUM_PARAM` as a 2-piece template you can "turn off" by setting it to nothing later. Interesting! I used this strategy in the grammar DSLs before. There I named these interfaces something like `SELECTING_COLUMNS_PART` and I added some `// comment` to the interface about how to treat this argument. Maybe it's useful to keep using this naming convention.